### DISCUSSION TITLE
tailwind.config.js doesn't take immediate effect

### DISCUSSION THREAD

**Discussion by reneezhou** [tailwindlabs/tailwindcss#2692] [Help] [answered]
Hi, 

Thanks to the amazing Tailwind team that created this amazing tool! I'm loving it. 🙏🏼

I'm using tailwind in combination with flask. My problem is I couldn't get my configuration behaving in the correct manner in 2 ways. 
1. Some extensions I made in tailwind.config.js would only take effect after I restarted my OS (macOS Catalina). 
2. I have not been able to use the @apply to build my own customized class. 
[I didn't experiment as described in 1. => create a class -> npm run build -> restart OS-> try and apply customized class, to see if it works.]

I checked the styles.css that was outputted every time after I ran npm run build - it did reveal changes I made. However, the effects were never immediate. It is a strange behavior. 
[I've also tried directly editing the default (full) tailwind.config.js => npm run build. Same, changes revealed in outputted css file but this didn't allow changes to take immediate effect. ]

When I started my project, I followed tailwind's official installation guide to set up tailwind. [https://tailwindcss.com/docs/installation](url)

Then I found issues as described above, I then followed this guide to set up my env again, added postcss & autoprefixer. [https://tailwindcss.com/course/setting-up-tailwind-and-postcss](url) 
During this time, I've run `npm cache clean --force`, which I'm not sure if it has potentially damaged anything. 

Here is the list of info of my tools and env:

- VSCode 1.50.1
- Tailwind CSS IntelliSense 0.4.3
- Python 3.8.6 (I build my projects in venv)
- Flask 1.1.2
- npm 6.14.8
- tailwind 1.9.6 [the result from `npm show tailwind`]

package.json
`{
  "main": "postcss.config.js",
  "scripts": {
    "build": "postcss src/styles.css -o .../static/styles.css"
  },
  "dependencies": {
    "autoprefixer": "^10.0.1",
    "postcss-cli": "^8.1.0",
    "tailwindcss": "^1.9.5"
  },
  "devDependencies": {
    "postcss": "^8.1.2"
  }
}
`

postcss.config.js
`module.exports = {
    plugins: [
        require('tailwindcss'),
        require('autoprefixer'),
    ]
}`

Thanks for your help in any case! Would love to have a full experience in this powerful & beautiful tool! 🙏🏼

**Comment by MartinHughes-BPC** [comment, marked as answer]
Hi Renee,

This sounds like a caching issue rather than a Tailwind issue.  Have a look at this StackOverflow answer to see if it helps: https://stackoverflow.com/questions/61740968/why-does-my-flask-app-not-render-css-if-i-dont-clear-my-cache

Best wishes,
Martin.

**Reply by reneezhou** [reply]
Hi Martin, 

Thank you so much! This has fixed my issue. 
Now my experience & productivity can be brought up to another level because of you! 🙏🏼 🌸

Cheers, 
Renee.
